File: stringArrayDuplicates.py
def dup(arry):
    result = []
    for word in arry:
        newWord = word[0]
        for i in range(len(word)):
            if newWord[-1] != word[i]:
                newWord += word[i]
        result.append(newWord)
    return result

print(dup(["ccooddddddewwwaaaaarrrrsssss","piccaninny","hubbubbubboo"]),['codewars','picaniny','hubububo'])
print(dup(["abracadabra","allottee","assessee"]),['abracadabra','alote','asese'])
print(dup(["kelless","keenness"]), ['keles','kenes'])
print(dup(["Woolloomooloo","flooddoorroommoonlighters","chuchchi"]), ['Wolomolo','flodoromonlighters','chuchchi'])
print(dup(["adanac","soonness","toolless","ppellee"]), ['adanac','sones','toles','pele'])
print(dup(["callalloo","feelless","heelless"]), ['calalo','feles','heles'])
print(dup(["putteellinen","keenness"]), ['putelinen','kenes'])
print(dup(["kelless","voorraaddoosspullen","achcha"]), ['keles','voradospulen','achcha'])

File: stringArrayDuplicates.js
function dup(array) {
    let result = []

    for (let i = 0; i < array.length; i++) {
        let word = array[i]
        let newWord = word.substr(0, 1)
        for (let j = 0; j < word.length; j++) {
            if (newWord[newWord.length - 1] != word.substr(j, 1))
                newWord += word.substr(j, 1)
        }
        result.push(newWord)
    }
    return result
}

console.log(dup(["ccooddddddewwwaaaaarrrrsssss","piccaninny","hubbubbubboo"]),'\n',['codewars','picaniny','hubububo'])
console.log(dup(["abracadabra","allottee","assessee"]),'\n',['abracadabra','alote','asese'])
console.log(dup(["kelless","keenness"]), '\n',['keles','kenes'])
console.log(dup(["Woolloomooloo","flooddoorroommoonlighters","chuchchi"]), '\n',['Wolomolo','flodoromonlighters','chuchchi'])
console.log(dup(["adanac","soonness","toolless","ppellee"]), '\n',['adanac','sones','toles','pele'])
console.log(dup(["callalloo","feelless","heelless"]), '\n',['calalo','feles','heles'])
console.log(dup(["putteellinen","keenness"]), '\n',['putelinen','kenes'])
console.log(dup(["kelless","voorraaddoosspullen","achcha"]), '\n',['keles','voradospulen','achcha'])